### PR TITLE
AutoYaST support for the OnlineOnly installation medium (jsc#SLE-7214)

### PR DESCRIPTION
**Pull request by lslezak**
- https://trello.com/c/nkA4Os8Y/1377-5-autoyast-installation-using-the-online-media
- https://jira.suse.com/browse/SLE-7214
- Tested manually
- 4.2.13

### CODE REVIEW

File: src/clients/scc_auto.rb [hunk 1]
```diff
@@ -212,6 +223,35 @@ def write
...
+      # import the GPG keys before refreshing the repositories
+      Yast::Packages.ImportGPGKeys
+
+      products = Y2Packager::ProductControlProduct.products
+      ay_product = Profile.current.fetch("software", {}).fetch("products", []).first
+
+      if !ay_product
+        # TRANSLATORS: error message, %s is the XML path, e.g. "software/products"
+        Report.Error(_("Missing product specification in the %s section") % "software/products")
+        return false
```
Comment (jreidinger) [inline src/clients/scc_auto.rb]: I prefer if specially designed class for autoyast issues will be used. I think this one with fatal severity is suitable https://github.com/yast/yast-autoinstallation/blob/master/src/lib/autoinstall/autoinst_issues/missing_value.rb

File: src/clients/scc_auto.rb [hunk 2]
```diff
@@ -212,6 +223,35 @@ def write
...
+      end
+
+      control_product = products.find { |p| p.name == ay_product }
+
+      if !control_product
+        # TRANSLATORS: error message, %s is a product ID, e.g. "SLES"
+        Report.Error(_("Product %s not found") % ay_product)
```
Comment (jreidinger) [inline src/clients/scc_auto.rb]: and here issue with invalid value.

File: src/clients/scc_auto.rb (hunk 1 above)
Comment (lslezak) [reply src/clients/scc_auto.rb]: Unfortunately the AY side expects only booleans [here](https://github.com/yast/yast-autoinstallation/blob/master/src/lib/autoinstall/autosetup_helpers.rb#L88-L95). Supporting custom AY issue would require bigger changes.

I'd keep it simple for now.

File: src/clients/scc_auto.rb (hunk 2 above)
Comment (lslezak) [reply src/clients/scc_auto.rb]: :question: :question: :question: Um, what exactly?

Comment (jreidinger) [reply src/clients/scc_auto.rb]: I just mean different type of autoyast issue https://github.com/yast/yast-autoinstallation/blob/master/src/lib/autoinstall/autoinst_issues/invalid_value.rb

but if it is not yet so ready, feel free to keep it as it is.

**Pull request merged**